packages/siopv2-oid4vp-op-auth/src/session/OpSession.ts

```diff
@@ -309,8 +309,12 @@ export class OpSession {
       const totalInputDescriptors = request.presentationDefinitions?.reduce((sum, pd) => {
         return sum + pd.definition.input_descriptors.length
       }, 0)
+      const totalVCs = args.verifiablePresentations?.reduce((sum, vp) => {
+        const uvp = CredentialMapper.toUniformPresentation(vp)
+        return sum + (uvp.verifiableCredential?.length ?? 0)
+      }, 0)
 
-      if (!request.presentationDefinitions || !args.verifiablePresentations || args.verifiablePresentations.length !== totalInputDescriptors) {
+      if (!request.presentationDefinitions || !args.verifiablePresentations || totalVCs !== totalInputDescriptors) {
```

Comment:
InputDescriptor does not equal 1 vc?

Reply:
I guess so, but each VP can have one or more VCs. So two VCs can be in one VP and a third can be in another VP.
"args.verifiablePresentations.length !== totalInputDescriptors" is not handling that.